src/ucd-tools/src/case.c

```diff
@@ -2841,7 +2841,7 @@ codepoint_t ucd_tolower(codepoint_t c)
 {
 	int begin = 0;
 	int end   = sizeof(case_conversion_data)/sizeof(case_conversion_data[0]);
-	while (begin <= end)
+       while (begin < end)
```

Comment:
1. This breaks the tests in the ucd-tools project:
```
cd src/ucd-tools
./autogen.sh
./configure --prefix=/usr
make
make check
```

2. The bug will also affect the other functions in that file.
3. The formatting of the file is different -- the line has changed to use spaces not tabs.

Reply:
> 1. This breaks the tests in the ucd-tools project:

Indeed, Made a better fix using `size` which remains constant instead of end

> 2. The bug will also affect the other functions in that file.

Fixed these as well

> 3. The formatting of the file is different -- the line has changed to use spaces not tabs.

Indeed, switched back to tabs

Reply:
Thanks, done